tests/integration/test_nginx_route.py

```diff
@@ -0,0 +1,192 @@
+    additional_hostnames: typing.Optional[str] = None,
+) -> str:
+    """Generate the src-overwrite config value for testing nginx-route relation."""
+    nginx_route_lib_path = "lib/charms/nginx_ingress_integrator/v0/nginx_route.py"
+    nginx_route_lib = Path(nginx_route_lib_path).read_text(encoding="utf8")
+    any_charm_script = Path("tests/integration/any_charm_nginx_route.py").read_text(
+        encoding="utf8"
+    )
+    nginx_route_config = {
+        "service_hostname": service_hostname,
+        "service_name": service_name,
+        "service_port": service_port,
+    }
+    if additional_hostnames:
+        nginx_route_config["additional_hostnames"] = additional_hostnames
+    any_charm_src_overwrite = {
+        "any_charm.py": any_charm_script,
+        "nginx_route.py": nginx_route_lib,
+        "nginx_route_config.json": json.dumps(nginx_route_config),
+    }
+    return json.dumps(any_charm_src_overwrite)
+
+
+def requests_get(url: str, host_header: str, retry_timeout: int = 120) -> requests.Response:
+    """Requests get, but will retry when the response status code is not 200."""
+    time_start = time.time()
+    while True:
+        response = requests.get(url, headers={"Host": host_header}, timeout=5)
+        if response.status_code == 200 or time.time() - time_start > retry_timeout:
+            return response
+
+
+@pytest_asyncio.fixture(scope="module")
+async def build_and_deploy(ops_test: OpsTest, run_action):
+    """build and deploy nginx-ingress-integrator charm.
+
+    Also deploy and relate an any-charm application for test purposes.
+
+    Returns: None.
+    """
+
+    assert isinstance(ops_test.model, Model)
+
+    async def build_and_deploy_ingress():
+        charm = await ops_test.build_charm(".")
+        assert isinstance(ops_test.model, Model)
+        return await ops_test.model.deploy(
+            str(charm), application_name="ingress", series="focal", trust=True
+        )
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            "any-charm",
+            application_name="any",
+            channel="beta",
+            config={"src-overwrite": gen_src_overwrite()},
+        ),
+        build_and_deploy_ingress(),
+    )
+    await ops_test.model.wait_for_idle()
+    await run_action(ANY_APP_NAME, "rpc", method="start_server")
+    relation_name = f"{INGRESS_APP_NAME}:nginx-route"
+    await ops_test.model.add_relation(f"{ANY_APP_NAME}:nginx-route", relation_name)
+    await ops_test.model.wait_for_idle()
+
+
+@pytest.mark.usefixtures("build_and_deploy")
+async def test_ingress_connectivity():
+    """
+    arrange: given charm has been built and deployed.
+    act: access ingress IP address with correct host name in HTTP headers.
+    assert: HTTP request should be forwarded to the application.
+    """
+    response = requests_get("http://127.0.0.1/ok", host_header="any")
+
+    assert response.text == "ok"
+    assert response.status_code == 200
+    assert (
```

Comment:
Is this assert described in the docstring? Perhaps worth a comment/ description?